src/games/ecards.py

```diff
@@ -0,0 +1,275 @@
+    wait_duration = 6
+    min_players = 2
+    cog_help = """
+    EGameFactory for a game where one player is given a prompt,
+    the other players select an answer from a set of safety answers,
+    and the head player selects their favourite answer.
+    """
+    has_scrape = True
+
+    file_prompts = "data/elash_prompts_{gid}.txt"
+    file_safeties = "data/elash_safeties_{gid}.txt"
+
+    def __init__(self, context):
+        super().__init__(context, __name__)
+
+        # instance property of all prompts
+        self.prompts = None
+        # instance property of all safety "cards"
+        self.safeties = None
+
+    async def start(self):
+        """
+        Method for starting the E Card game.
+        Reads in prompts, puts the deck in a particular state, and runs the game.
+        """
+
+        # check if prompts
+        if self.prompts == self.safeties == None:
+            # info message
+            self.logging.info(f"Reading prompts and safety files for {self.guild.id}.")
+            await self.channel.send(
+                embed=self.embed(f"Reading prompts and safety files...")
+            )
+
+            # read files
+            try:
+                self.prompts = self._read_file(self.file_prompts)
+                self.safeties = self._read_file(self.file_safeties)
+            except Exception as e:
+                self.logging.error(f"Error reading files {e}")
+                # early exit
+                return await self.channel.send(
+                    embed=self.embed(f"Error reading files\n```\n{e}\n```")
+                )
+
+            self.logging.info(
+                f"Read in {len(self.prompts)} prompts and {len(self.safeties)} safeties."
+            )
+
+        # create deck orders
+        prompt_deck = random.sample(self.prompts, len(self.prompts))
+        answer_deck = random.sample(self.safeties, len(self.safeties))
+
+        # create starting hands
+        hands = {pid: [answer_deck.pop() for _ in range(5)] for pid in self.players}
+
+        # run a round for each player
+        for pid in self.players:
+            self.logging.info(
+                f"new ecards round with leader player {self.players[pid]}"
+            )
+            await self.execute_round(pid, prompt_deck.pop(), hands)
+            ECards.refill_hands(hands, answer_deck)
+            asyncio.sleep(self.wait_duration)
+        return await self.scoreboard()
+
+    @staticmethod
+    def refill_hands(hands: dict, answer_deck: list):
+        """
+        Refills the players' hands from a deck of answers after a round.
+        Objects are passed by reference and modifications made in place, hence no need to return.
+
+        :param hands: a dictionary mapping player index to player hand
+        :param answer_deck: a deck of answer cards to fill hands with
+        """
+        for key in hands:
+            while len(hands[key]) < 5:
+                hands[key].append(answer_deck.pop())
```

Comment:
Do you want to bother checking if `answer_deck` is empty? I mean, there's so many "cards" I don't ever see it being an issue, but just to mention it.

Reply:
Planning on adding another copy of the shuffled decks to the end whenever they're running low in a future branch, just for the edge case of a game that continues so long.

Reply:
Okay, I'll resolve this comment then!